README.md

```diff
@@ -16,25 +18,150 @@ let s: &str = ".." ;
+Here is a little preview of what you can do with deserr:
+
+
+Let's say I sent this payload to update my [meilisearch](https://docs.meilisearch.com/reference/api/settings.html#settings) settings:
+```json
+{
+  "filterableAttributes": ["doggo.age", "catto.age"],
+  "sortableAttributes": ["uploaded_at"],
+  "typoTolerance": {
+    "minWordSizeForTypos": {
+      "oneTypo": 1000, "twoTypo": 80
+    },
+    "enabled": true
+  },
+  "displayedAttributes": ["*"],
+  "searchableAttributes": ["doggo.name", "catto.name"]
+}
+```
+
+#### With serde
+
+With serde, we don't have much customization; this is the typical kind of message we would get in return:
+
+```json
+{
+  "message": "Json deserialize error: invalid value: integer `1000`, expected u8 at line 6 column 21",
+  "code": "bad_request",
+  "type": "invalid_request",
+  "link": "https://docs.meilisearch.com/errors#bad_request"
+}
+```
+
+##### The message
+
+> Json deserialize error: invalid value: integer `1000`, expected u8 at line 6 column 21
+
+- The message uses the word `u8`, which definitely won't help a user who doesn't know rust or is unfamiliar with types.
+- The location is provided in terms of lines and columns. While this is generally good, when most of your users
+  read this message in their terminal, it doesn't actually help much.
+
+#### The rest of the payload
+
+Since serde returned this error, we cannot know what happened or on which field it happened. Thus, the best we
+can do is generate a code `bad_request` that is common for our whole API. We then use this code to generate
+a link to our documentation to help our users. But such a generic link does not help our users because it
+can be thrown by every single route of meilisearch.
+
+#### With deserr
+
+```json
+{
+  "message": "Invalid value at `.typoTolerance.minWordSizeForTypos.oneTypo`: value: `1000` is too large to be deserialized, maximum value authorized is `255`",
+  "code": "invalid_settings_typo_tolerance",
+  "type": "invalid_request",
+  "link": "https://docs.meilisearch.com/errors#invalid-settings-typo-tolerance"
+}
+```
+
+##### The message
+
+> Invalid value at `.typoTolerance.minWordSizeForTypos.oneTypo`: value: `1000` is too large to be deserialized, maximum value authorized is `255`
+
+- We get a more human-readable location; `.typoTolerance.minWordSizeForTypos.oneTypo`. It gives us the faulty field.
+- We also get a non-confusing and helpful message this time; it explicitly tells us that the maximum value authorized is `255`.
+
+##### The rest of the payload
+
+Since deserr called one of our functions in the process, we were able to use a custom error code + link to redirect
+our user to the documentation specific to this feature and this field.
+
+#### More possibilities with deserr that were impossible with serde
+
+##### Adding constraints on multiples fields
+
+In meilisearch, there is another constraint on this `minWordSizeForTypos`, the `twoTypo` field **must be** greater than
```

Comment:
Maybe you could add the version of Meilisearch to avoid having to update this line if Meilisearch changes its design.

```suggestion
In Meilisearch, there is another constraint on this `minWordSizeForTypos`, the `twoTypo` field **must be** greater than
```

Not sure my suggestion of capitalizing meilisearch should be taken into account. If it does, it should be changed everywhere.